plugin/core/sessions.py

```diff
@@ -1677,14 +1680,22 @@ def run_async() -> None:
 
             sublime.set_timeout_async(run_async)
             return Promise.resolve(None)
+        if is_refactoring:
+            self._is_executing_refactoring_command = True
         # TODO: Our Promise class should be able to handle errors/exceptions
-        return Promise(
+        execute_command = Promise(
             lambda resolve: self.send_request(
                 Request("workspace/executeCommand", command, None, progress),
                 resolve,
                 lambda err: resolve(Error(err["code"], err["message"], err.get("data")))
             )
         )
+        if is_refactoring:
+            execute_command.then(lambda _: self._reset_is_executing_refactoring_command())
```

Comment:
And a small style simplification: move the other line to here too and remove the two lines from above
```suggestion
        if is_refactoring:
            self._is_executing_refactoring_command = True
            execute_command.then(lambda _: self._reset_is_executing_refactoring_command())
```

Feel free to merge afterwards